### ISSUE TITLE
Confiuration not applied before docker-entrypoint-initdb.d scripts are applied

### ISSUE THREAD

**Issue by madsbuch** [supabase/postgres#291]
# Bug report
Per the postgres docker image it should be possible to place initial sql files in the `docker-entrypoint-initdb.d` folder for them to execute when postgres starts. When this is done in the supabase/postgres image default users are non-existing on the execution time wy this function does not work properly.

## To Reproduce

1. Start a development instance of the normal supabase infrastructure
2. Populate the database with some data
3. export the data using the `pg_dump` util
4. Start a database in a docker-compose environment
    ```
      database:
        image: supabase/postgres:latest
        ports:
          - 5432:5432
        command: postgres -c config_file=/etc/postgresql/postgresql.conf 
        volumes:
          - the-dump.sql:/docker-entrypoint-initdb.d/the-dump.sql
        environment:
          POSTGRES_PASSWORD: postgres
    ```
5. Observe following error:
   ```
   database_1  | [local] 2022-09-07 12:46:16.815 UTC [70] postgres@postgres ERROR:  role "supabase_admin" does not exist
   database_1  | [local] 2022-09-07 12:46:16.815 UTC [70] postgres@postgres STATEMENT:  ALTER SCHEMA auth OWNER TO 
   supabase_admin;
   database_1  | psql:/docker-entrypoint-initdb.d/preview_data.sql:26: ERROR:  role "supabase_admin" does not exist
   preview-infrastructure_database_1 exited with code 3
   ```

## Expected behavior

I would expect the database to start and populate the database with the contents of the dump.

## System information

- OS: PopOs 22.04

```
❯❯❯ docker-compose -v
docker-compose version 1.29.2, build unknown
❯❯❯ docker -v
Docker version 20.10.17, build 100c701
❯❯❯ supabase -v
1.4.2
```

**Comment by madsbuch** [comment]
TL;DR: Apply appropriate initialization. If the goal is to have it working like Supabase, take inspiration from here: https://github.com/supabase/supabase/tree/master/docker/volumes/db/init

I am letting myself out
